src/effects/EffectUI.cpp

```diff
@@ -882,16 +883,24 @@ void EffectUIHost::OnUserPreset(wxCommandEvent & evt)
 {
    int preset = evt.GetId() - kUserPresetsID;
    
-   mEffectUIHost.GetDefinition()
-      .LoadUserPreset(mEffectUIHost.GetUserPresetsGroup(mUserPresets[preset]));
+   // Make mutable copy
+   auto settings = mpAccess->Get();
```

Comment:
After object copied it's contents will be discarded by `LoadUserPreset`

Reply:
But observe that `mpAccess->Set(std::move(settings))` follows.

Reply:
The reading, modifying, and rewriting of EffectSettings stored elsewhere is done this long way using EffectSettingsAccess.  The last commit in this branch introduces another implementation of that abstract class, which handles inter-thread communication of changes of settings.